Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -559,8 +664,104 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
         return false;
     }
 
+#if OS(ANDROID)
+    return sendHardwareBuffers(WTFMove(hardwareBuffers));
+#else
+    return true;
+#endif
+}
+
+#if OS(ANDROID)
+bool Connection::sendHardwareBuffers(Deque<RefPtr<AHardwareBuffer>, 2>&& hardwareBuffers)
+{
+    RELEASE_ASSERT(m_outgoingHardwareBuffers.isEmpty());
+    m_outgoingHardwareBuffers = WTFMove(hardwareBuffers);
+    return sendOutgoingHardwareBuffers();
+}
```

Comment:
Why do we need this function? I guess we could do this in sendOutputMessage since it's only called there.

Reply:
For me it helps understand better how the code works: one prepares to start the process, does checks, etc., and then calls the other that actually does it, and the second one is the only function that gets re-scheduled in case of `EAGAIN`/`EWOULDBLOCK`.

The above is only a personal preference, so I'll move this as suggested.